docs/DeveloperGuide.md

```diff
@@ -4,26 +4,162 @@
+description. Each attribute can be obtained using public get methods, and the attribute quantity can be set using the public set method.
+
+
+## Implementation
+### Regex Class
+#### Description
+The Regex Class is created to simplify the use of regular expressions in Java.
+![Regex Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T09-4/tp/master/docs/diagrams/Regex.puml)
+The Class Diagram above shows the different variables and methods associated with the regex class.
+This class takes in the regular expression as a string and the string to be matched in the class contructor, and stores the results
+in the private variable `groupValues` which is a Hash Map where the key would be the capture group name and the value as the matched results.
+
+#### Operation
+![Regex Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T09-4/tp/master/docs/diagrams/RegexMatching.puml)
+The above diagram shows the inner workings of how the Regex Class will work. Here are the steps taken by the Regex Class:
+1. findMatch() called when instantiating groupValues variable
+2. findGroups() called within findMatch() to find the capture group names within the regex string
+3. regexMatching() will then be called within findMatch() which will use Java's base Pattern and Matcher Class, and returns a Matcher Object
+4. findMatch() will then use the Matcher object to fill in the HashMap of <key, value> pair of <capture group name, regex matching>
+5. findMatch() will then return the HashMap object to groupValues variable
+
+#### Example of Usage
+```
+String regex = "id/(?<id>\\d*) n/(?<name>.*) q/(?<qty>\\d*) d/(?<desc>\\.*)";
+Regex regexMatch = new Regex(userInput, regex);
+HashMap<String, String> matches = regexMatch.getGroupValues();
+String id = matches.get("id");
+String name = matches.get("name");
+String quantity = matches.get("qty");
+String description = matches.get("desc");
+```
+### Commands Class
+#### Description
+The Commands Class is the class which contains all the available commands that can be used in the Simplst CLI application.
+![Commands Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T09-4/tp/master/docs/diagrams/CommandsClass.puml)
+
+More about how each feature is run can be seen in each of the methods that can be found below.
+
+### View Good Method
+#### Description
+View Good belongs as part of the Commands Class. It is used when a user would like to know more information about an inventory item has an item id linked to it in the warehouse.
+
+#### Operation
+![View Good Sequence Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T09-4/tp/master/docs/diagrams/viewGood.puml)
```

Comment:
`User` is not a class, so the colon before `User` should not be included. Consider removing it.
![image](https://user-images.githubusercontent.com/60597985/160963256-3ede9091-d721-4a56-8371-884802eede86.png)